src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -4233,15 +4282,365 @@ module Mutations = struct
                           let next_tm_next = Time.add tm_next wait_span in
                           go ((ndx + 1) mod num_senders) next_tm_next
                       in
+                      [%log info] "Starting payment scheduler with handle %s"
+                        (Uuid.to_string uuid) ;
                       let tm_next = Time.add tm_start wait_span in
                       don't_wait_for @@ go 0 tm_next ;
                       Ok (Uuid.to_string uuid) ) )
 
-    let stop_payments =
-      io_field "stopPayments"
+    let account_of_id id ledger =
+      Mina_ledger.Ledger.location_of_account ledger id
+      |> Option.value_exn
+      |> Mina_ledger.Ledger.get ledger
+      |> Option.value_exn
+
+    let id_of_kp (kp : Signature_lib.Keypair.t) =
+      Account_id.create
+        (Signature_lib.Public_key.compress kp.public_key)
+        Token_id.default
+
+    let account_of_kp (kp : Signature_lib.Keypair.t) ledger =
+      account_of_id (id_of_kp kp) ledger
+
+    let deploy_zkapps ~mina ~ledger
+        ~(fee_payer_keypair : Signature_lib.Keypair.t)
+        ~(account_creator_keypair : Signature_lib.Keypair.t)
+        ~constraint_constants ~logger keypairs =
+      let fee_payer_account = account_of_kp fee_payer_keypair ledger in
+      let account_creator = account_of_kp account_creator_keypair ledger in
+      let fee_payer_nonce = ref fee_payer_account.nonce in
+      let account_creator_nonce = ref account_creator.nonce in
+      Deferred.List.iter keypairs ~f:(fun kp ->
+          let spec =
+            { Transaction_snark.For_tests.Deploy_snapp_spec.sender =
+                (account_creator_keypair, !account_creator_nonce)
+            ; fee = Currency.Fee.of_mina_string_exn "1.0"
+            ; fee_payer = Some (fee_payer_keypair, !fee_payer_nonce)
+            ; amount = Currency.Amount.of_mina_string_exn "20000.0"
+            ; zkapp_account_keypairs = [ kp ]
+            ; memo = Signed_command_memo.empty
+            ; new_zkapp_account = true
+            ; snapp_update = Account_update.Update.dummy
+            ; preconditions = None
+            ; authorization_kind = Account_update.Authorization_kind.Signature
+            }
+          in
+          let zkapp_command =
+            Transaction_snark.For_tests.deploy_snapp ~constraint_constants
+              ~default_permissions:true spec
+          in
+          let rec go () =
+            match%bind send_zkapp_command mina zkapp_command with
+            | Ok _ ->
+                fee_payer_nonce := Account.Nonce.succ !fee_payer_nonce ;
+                account_creator_nonce :=
+                  Account.Nonce.succ !account_creator_nonce ;
+                [%log info]
+                  "Successfully send out zkApp $command that creates a zkApp \
```

Comment:
Interpolating the zkApp in the log message will make it too verbose, I think. So I think removing `$command` is better.

Reply:
maybe "submitted" is better than "send out", because so far, the txn is just added to the transaction pool